darts/models/forecasting/torch_forecasting_model.py

```diff
@@ -1872,8 +1872,9 @@ def load_weights_from_checkpoint(
         )
 
         # pl_forecasting module saves the train_sample shape, must recreate one
+        np_dtype = np.float32 if ckpt["model_dtype"] == torch.float32 else np.float64
```

Comment:
I'd suggest creating a mapping between the torch and numpy dtypes at the top of this file. With this we can avoid/catch some breaking changes (e.g. if we ever support other dtypes such as float16 and forget about this line).

```
TORCH_NP_DTYPES = {
    torch.float32: np.float32,
    torch.float64: np.float64,
}

...

mock_train_sample = [
    np.zeros(sample_shape, dtype=TORCH_NP_DTYPES[ckpt["model_dtype"]]) if sample_shape else None
    for sample_shape in ckpt["train_sample_shape"]
]

Reply:
also add float16

```python
TORCH_NP_DTYPES = {
    torch.float16: np.float16,
    torch.float32: np.float32,
    torch.float64: np.float64,
}
```